meilisearch-release-changelog/v1.11.0.md

```diff
@@ -0,0 +1,208 @@
+* When using the semantic or the hybrid search, `hybrid.embedder` is now a **mandatory** parameter in `GET and POST /indexes/{:indexUid}/search`
+* As a consequence, it is now **mandatory** to pass `hybrid` even for full-vector search (with only `vector` and not `q`)
+* `embedder` is now a **mandatory** parameter in `GET and POST /indexes/{:indexUid}/similar`
+* Ignore non-zero `semanticRatio` when `vector` is passed but not `q`: a semantic search will be performed.
+
+### Addition & improvements
+
+* The default model for OpenAI is now `text-embedding-3-small` instead of `text-embedding-ada-002`.
+* Limit the maximum length of a rendered document template: when the source of an embedder is set to `huggingFace`, `openAi`, `rest` or `ollama`, then `documentTemplateMaxBytes` is now available as an optional parameter. This parameter describes the number of bytes in which the rendered document template text should fit when trying to embed a document. Longer texts are truncated to fit.
+* Add the ability to query whether a field is searchable in `documentTemplate`: use `field.is_searchable`:  `true` if the field is a searchable attribute, otherwise `false`.
+* The new document template changed to
+  * filter out fields that are not searchable
+  * filter out fields that are nil or missing for the current document
+
+v1.10 and before:
+```
+{% for field in fields %}
+{{ field.name }}: {{ field.value }}\n
+{% endfor %}
+```
+
+From v1.11:
+```
+{% for field in fields %}
+{% if field.is_searchable and not field.value == nil %}
+{{ field.name }}: {{ field.value }}\n
+{% endif %}
+{% endfor %}
+```
+
+This last change is backward compatible. Embedders that use the v1.10 document template will keep using it. The new default document template will only be applied to newly added embedders.
+
+## Speed up indexing of your vector database with binary quantization
+
+The binary quantization converts your embeddings of floating point numbers (using 32 bits) into embeddings of boolean values (using 1 bit).
+In our benchmarks, it reduced the size of the database by a factor of 10, and divided the indexing time by a factor of 6 without much impact on the search time.
+
+> [!WARNING]
+> Enabling this feature will update all of your vectors to contain only `1`s or `-1`s, impacting the relevancy significantly.
+> Once enabled, **there is no coming back**. Before trying it out on your main index, you should try the feature on a smaller index to ensure it fits your use case.
+
+Although this feature degrades the relevancy of the semantic searches, it showed great results on collections with many dimensions (> 1300) and many points (more than 100000).
+
+Enable the binary quantization by updating the settings of your embedder with:
+```bash
+curl \
+  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "embedders": {
+      "image2text": {
+        "binaryQuantized": true
+      }
+    }
+  }'
+```
```

Comment:
````suggestion
v1.11 introduces a new embedder option, `binaryQuantized`:

```bash
curl \
  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
  -H 'Content-Type: application/json' \
  --data-binary '{
    "embedders": {
      "image2text": {
        "binaryQuantized": true
      }
    }
  }'
```

Enable binary quantization to convert embeddings of floating point numbers into embeddings of boolean values. This will negatively impact the relevancy of AI-powered searches, but significantly improve performance in large collections with more than 100 dimensions.

In our benchmarks, this reduced the size of the database by a factor of 10, and divided the indexing time by a factor of 6 with little impact on search times.

> [!WARNING]
> Enabling this feature will update all of your vectors to contain only `1`s or `-1`s, impacting the relevancy significantly.
> 
> **You cannot revert this option once you enable it**. Before setting `binaryQuantized` to `true`, Meilisearch recommends testing it in a smaller or duplicate index in a development environment.
````